File: WebSharperMobileWP7EmptyApp/www/Scripts/IntelliFactory.WebSharper.Mobile.Default/WebSharper.js
(function()
{
 var global=this;
 var Mobile,WebSharper,localStorage;
 IntelliFactory.WebSharper.Runtime.Declare({IntelliFactory:{WebSharper:{Mobile:{DefaultMobileProvider:{}}}}});
 Mobile=function()
 {
  return IntelliFactory.WebSharper.Mobile;
 };
 WebSharper=function()
 {
  return IntelliFactory.WebSharper;
 };
 localStorage=function()
 {
  return window.localStorage;
 };
 (Mobile()).DefaultMobileProvider=(WebSharper()).Runtime.Class(function()
 {
  var _this=this;
  var _,__1,c,matchValue;
  matchValue={$:0};
  if(matchValue.$==1)
   {
    c=matchValue.$0;
    _=[];
    __1=void c.apply(_this,_);
   }
  __1;
 },null,{Alert:function(s)
 {
  return alert(s);
 },Log:function(s)
 {
  var _;
  _=console;
  return console.log.call(_,s);
 },GetLocation:function()
 {
  return{Lat:0,Long:0};
 },GetAcceleration:function()
 {
  return{X:0,Y:0,Z:-9.81};
 },GetPhotoFromCamera:function()
 {
  var builder_;
  builder_=(WebSharper()).Control.ExtraTopLevelOperators.Do;
  return builder_.Delay(function()
  {
   return builder_.Return("");
  });
 },StorageLoad:function(k)
 {
  var _,_1,__1,__2,x;
  try
  {
   _=localStorage();
   __2=_.getItem(k);
   __1=function(_arg1)
   {
    return _arg1===null?"":_arg1;
   };
   _1=__1(__2);
  }
  catch(x)
  {
   _1="";
  }
  return _1;
 },StorageStore:function(k,v)
 {
  var _;
  _=localStorage();
  return _.setItem(k,v);
 },JsonStorageLoad:function(k)
 {
  var _this=this;
  var _,__1,__3,__4,__5,__6;
  __1=_this.StorageLoad(k);
  _=function(_arg2)
  {
   return _arg2===""?"[]":_arg2;
  };
  __4=_(__1);
  __3=function(arg00)
  {
   var __2;
   __2=JSON;
   return JSON.parse.call(__2,arg00);
  };
  __6=__3(__4);
  __5=function(value)
  {
   return value;
  };
  return __5(__6);
 },JsonStorageStore:function(k,v)
 {
  var _this=this;
  var _,arg10;
  _=JSON;
  arg10=JSON.stringify.call(_,v);
  return _this.StorageStore(k,arg10);
 }});
 (Mobile()).Mobile=new(Mobile()).DefaultMobileProvider();
}());

File: if-ws-mobile/WebSharper.fs
// $begin{copyright}
//
// This file is confidential and proprietary.
//
// Copyright (c) IntelliFactory, 2004-2011.
//
// All rights reserved.  Reproduction or use in whole or in part is
// prohibited without the written consent of the copyright holder.
//-----------------------------------------------------------------
// $end{copyright}

module IntelliFactory.WebSharper.Mobile

open IntelliFactory.WebSharper
open IntelliFactory.WebSharper.Html
open IntelliFactory.WebSharper.Html5
open IntelliFactory.WebSharper.Json

type Location =
    {
        Lat : float
        Long : float
    }

type Acceleration =
    {
        X : float
        Y : float
        Z : float
    }

type Image = string

type IMobileProvider =

    /// Displays an alert.
    abstract member Alert : string -> unit

    /// Logs a message.
    abstract member Log : string -> unit

    /// Returns the current location of the device.
    abstract member GetLocation : unit -> Location

    /// Returns the current acceleration of the device.
    abstract member GetAcceleration : unit -> Acceleration

    /// <summary>Returns a photo from the camera.</summary>
    abstract member GetPhotoFromCamera : unit -> Async<Image>

    /// Loads a value from local storage.
    abstract member StorageLoad : string -> string

    /// Stores a value to local storage.
    abstract member StorageStore : string -> string -> unit

    /// Loads a value from local storage.
    abstract member JsonStorageLoad<'a> : string -> 'a

    /// Stores a value to local storage.
    abstract member JsonStorageStore<'a> : string -> 'a -> unit

type private DefaultMobileProvider [<JavaScript>] () =

    [<Inline "alert($s)">]
    let bAlert (s : string) : unit = X

    [<Inline "console.log($s)">]
    let bLog (s : string) : unit = X

    interface IMobileProvider with

        [<JavaScript>]
        override this.Alert s = bAlert s

        [<JavaScript>]
        override this.Log s = bLog s

        [<JavaScript>]
        override this.GetLocation() : Location =
            { Lat = 0.; Long = 0. }

        [<JavaScript>]
        override this.GetAcceleration() : Acceleration =
            { X = 0.; Y = 0.; Z = -9.81 }

        [<JavaScript>]
        override this.GetPhotoFromCamera() =
            async { return "" }

        [<JavaScript>]
        override this.StorageLoad k =
            try
                Window.Self.LocalStorage.GetItem k
                |> function
                    | x when x = null -> ""
                    | x -> x
            with _ -> ""

        [<JavaScript>]
        override this.StorageStore k v =
            Window.Self.LocalStorage.SetItem (k, v)

        [<JavaScript>]
        override this.JsonStorageLoad k =
            let x =
                match (this :> IMobileProvider).StorageLoad k with
                | "" -> "[]"
                | x -> x
            unbox (Json.Parse x)

        [<JavaScript>]
        override this.JsonStorageStore k v =
            (this :> IMobileProvider).StorageStore k (Json.Stringify v)

[<JavaScript>]
let mutable Mobile : IMobileProvider =
    DefaultMobileProvider() :> _
